### PR TITLE
2750 schema descriptions

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Fixes part of #2750

Specify descriptions to all fields missing one instead of overriding pydantic methods

## Testing Checklist

* [ ] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by glossing over the UI, making sure those descriptions appear
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Comment by VakarisZ** [comment]
@ybasford we need help :D

**Review by mssalvatore — APPROVED** [review]
LGTM

File: monkey/common/agent_configuration/agent_sub_configurations.py [hunk 1]
```diff
@@ -191,7 +195,9 @@ class ExploitationConfiguration(MutableInfectionMonkeyBaseModel):
         :param exploiters: Configuration enabled exploiters
     """
 
-    options: ExploitationOptionsConfiguration = Field(title="Exploiters Options")
+    options: ExploitationOptionsConfiguration = Field(
+        title="Exploiters Options", description="Configure other options related to exploitation"
```
Comment (shreyamalviya) [inline monkey/common/agent_configuration/agent_sub_configurations.py:199]: Maybe
```suggestion
        title="Exploiters Options", description="Configure exploitation options shared by all exploiters"
```

File: monkey/common/agent_configuration/agent_sub_configurations.py [hunk 2]
```diff
@@ -221,5 +227,10 @@ class PropagationConfiguration(MutableInfectionMonkeyBaseModel):
         "Setting this to 0 will disable all scanning and exploitation.",
         default=2,
     )
-    network_scan: NetworkScanConfiguration = Field(title="Network analysis")
-    exploitation: ExploitationConfiguration = Field(title="Exploiters")
+    network_scan: NetworkScanConfiguration = Field(
+        title="Network analysis",
+        description="Configure the network analysis the Agents will perform",
```
Comment (shreyamalviya) [inline monkey/common/agent_configuration/agent_sub_configurations.py:232]:
```suggestion
        description="Configure the network analysis that the Agents will perform",
```

**Review by shreyamalviya — CHANGES_REQUESTED** [review]
Some suggestions

**Review by shreyamalviya — APPROVED** [review]

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/common/agent_configuration/agent_configuration.py [hunk 3]
```diff
@@ -15,6 +15,14 @@ class AgentConfiguration(MutableInfectionMonkeyBaseModel):
         "seconds)",
         default=30,
     )
-    credential_collectors: Tuple[PluginConfiguration, ...] = Field(title="Credential collectors")
-    payloads: Dict[str, Dict] = Field(title="Payloads")
-    propagation: PropagationConfiguration = Field(title="Propagation")
+    credential_collectors: Tuple[PluginConfiguration, ...] = Field(
+        title="Credential collectors",
+        description="Configure options related to the credential collection stage of the attack",
```
Comment (ybasford) [inline monkey/common/agent_configuration/agent_configuration.py:20]: Suggested change: 
Configure options for the attack’s credential collection stage

File: monkey/common/agent_configuration/agent_configuration.py [hunk 4]
```diff
@@ -15,6 +15,14 @@ class AgentConfiguration(MutableInfectionMonkeyBaseModel):
...
+    )
+    payloads: Dict[str, Dict] = Field(
+        title="Payloads", description="Configure payloads that Agents will execute"
+    )
+    propagation: PropagationConfiguration = Field(
+        title="Propagation",
+        description="Configure options related to the propagation stage of the attack",
```
Comment (ybasford) [inline monkey/common/agent_configuration/agent_configuration.py:27]: Suggested change: 
Configure options for the attack’s propagation stage

File: monkey/common/agent_configuration/agent_sub_configurations.py [hunk 5]
```diff
@@ -151,8 +151,12 @@ class NetworkScanConfiguration(MutableInfectionMonkeyBaseModel):
...
-    icmp: ICMPScanConfiguration = Field(title="Ping scanner")
+    tcp: TCPScanConfiguration = Field(
+        title="TCP scanner", description="Configure TCP scanning options"
+    )
+    icmp: ICMPScanConfiguration = Field(
+        title="Ping scanner", description="Configure ICMP scanning options"
+    )
     fingerprinters: Tuple[PluginConfiguration, ...] = Field(
         title="Fingerprinters",
         description="Fingerprint modules collect info about external "
```
Comment (ybasford) [inline monkey/common/agent_configuration/agent_sub_configurations.py:162]: Suggested change: 
Fingerprint modules that collect info about external services

File: monkey/common/agent_configuration/agent_sub_configurations.py [hunk 6]
```diff
@@ -177,7 +181,7 @@ class ExploitationOptionsConfiguration(MutableInfectionMonkeyBaseModel):
     """
 
     http_ports: Tuple[NetworkPort, ...] = Field(
-        title="HTTP Ports",
+        title="HTTP ports",
         description="List of ports the monkey will check if are being used for HTTP",
```
Comment (ybasford) [inline monkey/common/agent_configuration/agent_sub_configurations.py:185]: Suggested change: 
A list of ports that the Monkey checks for HTTP traffic
@VakarisZ let me know if "HTTP traffic" is the right term here

File: monkey/common/agent_configuration/agent_sub_configurations.py [hunk 7]
```diff
@@ -192,7 +196,10 @@ class ExploitationConfiguration(MutableInfectionMonkeyBaseModel):
         :param exploiters: Configuration enabled exploiters
     """
 
-    options: ExploitationOptionsConfiguration = Field(title="Exploiters Options")
+    options: ExploitationOptionsConfiguration = Field(
+        title="Exploiters options",
+        description="Configure exploitation options shared by all exploiters",
```
Comment (ybasford) [inline monkey/common/agent_configuration/agent_sub_configurations.py:201]: This text is good

File: monkey/common/agent_configuration/agent_sub_configurations.py [hunk 8]
```diff
@@ -222,5 +229,10 @@ class PropagationConfiguration(MutableInfectionMonkeyBaseModel):
         "Setting this to 0 will disable all scanning and exploitation.",
         default=2,
     )
-    network_scan: NetworkScanConfiguration = Field(title="Network analysis")
-    exploitation: ExploitationConfiguration = Field(title="Exploiters")
+    network_scan: NetworkScanConfiguration = Field(
+        title="Network analysis",
+        description="Configure the network analysis that the Agents will perform",
```
Comment (ybasford) [inline monkey/common/agent_configuration/agent_sub_configurations.py:234]: This text is good

File: monkey/common/agent_configuration/agent_sub_configurations.py (hunk 5 above)
Comment (VakarisZ) [reply monkey/common/agent_configuration/agent_sub_configurations.py:162]: I think we want to describe what fingerprinter plugins are. My suggestion is to do "Fingerprint plugins identify network services found".

**Comment by ybasford** [comment]
I agree with Vakaris

Yael Basford
Senior Technical Writer

[signature_3632670654]


Office:+1.617.444.1234
Cell:+1.978.868.8785

Akamai Technologies
145 Broadway
Cambridge, MA 02142


Connect with Us:

[signature_1409130784]<https://community.akamai.com/>[signature_3681041544]<http://blogs.akamai.com/>[signature_2795187136]<https://twitter.com/akamai>[signature_3330014064]<http://www.facebook.com/AkamaiTechnologies>[signature_272063351]<http://www.linkedin.com/company/akamai-technologies>[signature_4084934274]<http://www.youtube.com/user/akamaitechnologies?feature=results_main>



From: VakarisZ ***@***.***>
Reply-To: guardicore/monkey ***@***.***>
Date: Wednesday, March 1, 2023 at 10:30
To: guardicore/monkey ***@***.***>
Cc: "Basford, Yael" ***@***.***>, Mention ***@***.***>
Subject: Re: [guardicore/monkey] 2750 schema descriptions (PR #3017)


@VakarisZ commented on this pull request.

________________________________

In monkey/common/agent_configuration/agent_sub_configurations.py<https://urldefense.com/v3/__https:/github.com/guardicore/monkey/pull/3017*discussion_r1121341165__;Iw!!GjvTz_vk!ShZx-kf8KuaUbbCaTwpG-CAmVe-XwALLLTsDCfLvmrJgYz2MBO61U4OLu0awsatwa4nIXHOEfjjfJVge-YDtJ67Y$>:

> @@ -151,8 +151,12 @@ class NetworkScanConfiguration(MutableInfectionMonkeyBaseModel):

         :param targets: Configuration for targets to scan

     """



-    tcp: TCPScanConfiguration = Field(title="TCP scanner")

-    icmp: ICMPScanConfiguration = Field(title="Ping scanner")

+    tcp: TCPScanConfiguration = Field(

+        title="TCP scanner", description="Configure TCP scanning options"

+    )

+    icmp: ICMPScanConfiguration = Field(

+        title="Ping scanner", description="Configure ICMP scanning options"

+    )

     fingerprinters: Tuple[PluginConfiguration, ...] = Field(

         title="Fingerprinters",

         description="Fingerprint modules collect info about external "

I think we want to describe what fingerprinter plugins are. My suggestion is to do "Fingerprint plugins identify network services found".

File: monkey/common/agent_configuration/agent_sub_configurations.py (hunk 6 above)
Comment (VakarisZ) [reply monkey/common/agent_configuration/agent_sub_configurations.py:185]: Monkey doesn't check for traffic on these ports. It just tries to figure out whether these ports are serving HTTP

Comment (VakarisZ) [reply monkey/common/agent_configuration/agent_sub_configurations.py:185]: I've done "List of ports the Agent will check for using HTTP protocol", but this option will be short lived

Comment (ybasford) [reply monkey/common/agent_configuration/agent_sub_configurations.py:185]: "A list of ports the Monkey checks for HTTP services"
Is that better or am I still not nailing it?

Comment (ybasford) [reply monkey/common/agent_configuration/agent_sub_configurations.py:185]: Why short lived? Will this change any time soon?

Comment (ybasford) [reply monkey/common/agent_configuration/agent_sub_configurations.py]: Final: 
List of ports the Agent will check for using an HTTP protocol

**Pull request merged**